torch/_tensor_str.py

```diff
@@ -280,7 +280,9 @@ def _str(self):
             or (self.device.type == 'cuda' and torch.cuda.current_device() != self.device.index):
         suffixes.append('device=\'' + str(self.device) + '\'')
 
-    has_default_dtype = self.dtype in (torch.get_default_dtype(), torch.int64, torch.bool)
+    # TODO: add an API to map real -> complex dtypes
+    _default_complex_dtype = torch.cfloat if torch.get_default_dtype() == torch.float else torch.cdouble
```

Comment:
so if default dtype is half this is cdouble?

Reply:
good point! no it should be cdouble if double else cfloat, similar to the logic [here](https://github.com/pytorch/pytorch/blob/a44824c9eda0301bb497840904ac1c64f0c69b89/c10/core/DefaultDtype.cpp#L11)

Reply:
will update!